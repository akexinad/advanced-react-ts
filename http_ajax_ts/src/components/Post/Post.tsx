import React, { FC } from "react";
import { withRouter, RouteComponentProps } from "react-router-dom";

import { IPost } from "../../interfaces";

import styles from "./Post.module.css";

export interface PostProps extends RouteComponentProps {
    id: IPost["id"];
    title: IPost["title"];
    author: IPost["author"];
    clicked: (id: IPost["id"]) => void;
}

const Post: FC<PostProps> = ({ id, title, author, clicked }) => (
    <article className={styles.Post} onClick={() => clicked(id)}>
        <h1>{title}</h1>
        <div className={styles.Info}>
            <div className={styles.Author}>{author}</div>
        </div>
    </article>
);

export default withRouter(Post);
