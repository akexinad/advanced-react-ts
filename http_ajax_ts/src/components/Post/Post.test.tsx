import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import Post, { PostProps } from "./Post";



test("Post Component", () => {
    const id: PostProps["id"] = "abc123";
    const _clicked = (id: PostProps["id"]) => null;

    const component = renderer.create(
        <BrowserRouter>
            <Post
                id="abc123"
                title={"post title"}
                author={"post author"}
                clicked={() => _clicked(id)}
            />
        </BrowserRouter>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
