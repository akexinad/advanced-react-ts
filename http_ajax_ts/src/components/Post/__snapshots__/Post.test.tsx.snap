// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Component 1`] = `
<article
  className="Post"
  onClick={[Function]}
>
  <h1>
    post title
  </h1>
  <div
    className="Info"
  >
    <div
      className="Author"
    >
      post author
    </div>
  </div>
</article>
`;
